### PR TITLE
[show] configuration bus support

### PR DESCRIPTION
**Pull request by rtxm**

### CODE REVIEW

File: tests/test_batch.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+from copy import deepcopy
```
Comment (rtxm) [inline tests/test_batch.py:1]: this file is a left-over of a previous PR… :disappointed:

File: examples/ping-pika-conf.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+#
+# MIT License
+#
+# Copyright (c) 2018-2019 Groupe Allo-Media
```
Comment (merwan) [inline examples/ping-pika-conf.py:5]:
```suggestion
# Copyright (c) 2022 Groupe Allo-Media
```

File: src/eventail/async_service/pika/base.py [hunk 3]
```diff
@@ -1059,7 +1122,18 @@ def handle_returned_message(
...
+        """Code to execute once the business service is configured and comes online.
+
+        (to be implemented by subclasses)
+        """
+        pass
+
+    def on_unconfigured(self) -> None:
+        """This handler is called when the service needs to be configured before it comes online.
+
+        This callack is invoked only if there are configuration subscriptions (at least one).
```
Comment (merwan) [inline src/eventail/async_service/pika/base.py:1134]:
```suggestion
        This callback is invoked only if there are configuration subscriptions (at least one).
```

**Review by merwan — COMMENTED** [review]
Thanks to this PR I better understand how the lib behaves :+1:
When testing locally I had an issue: when testing `examples/ping-pika-conf.py`, the program never exits after doing a Ctrl-C before the application is configured. It is stuck forever... It seems like a bug to me, could you check on your side?

**Comment by rtxm** [comment]
> the program never exits after doing a Ctrl-C before the application is configured. It is stuck forever... It seems like a bug to me, could you check on your side?

Good catch! it's indeed a bug. Fix on the way.

**Pull request merged**